firestore-bigquery-export/firestore-bigquery-change-tracker/src/bigquery/schema.ts

```diff
@@ -76,44 +77,104 @@ export const documentPathParams = {
+    mode: "REQUIRED",
+    type: "STRING",
+    description:
+      "The full name of the changed document, for example, projects/collection/databases/(default)/documents/users/me).",
+  },
+  {
+    name: "operation",
+    mode: "REQUIRED",
+    type: "STRING",
+    description: "One of CREATE, UPDATE, IMPORT, or DELETE.",
+  },
+];
+
+const defaultViewSchemaFields = [
+  {
+    name: "timestamp",
+    mode: "NULLABLE",
+    type: "TIMESTAMP",
+    description:
+      "The commit timestamp of this change in Cloud Firestore. If the operation is IMPORT, this timestamp is epoch to ensure that any operation on an imported document supersedes the IMPORT.",
+  },
+  {
+    name: "event_id",
+    mode: "NULLABLE",
+    type: "STRING",
+    description:
+      "The ID of the most-recent document change event that triggered the Cloud Function created by the extension. Empty for imports.",
+  },
+  {
+    name: "document_name",
+    mode: "NULLABLE",
+    type: "STRING",
+    description:
+      "The full name of the changed document, for example, projects/collection/databases/(default)/documents/users/me).",
+  },
+  {
+    name: "operation",
+    mode: "NULLABLE",
+    type: "STRING",
+    description: "One of CREATE, UPDATE, IMPORT.",
+  },
+];
+
+export const SelectRawChangelogViewSchema = (type = "STRING") => {
+  if (type === "STRING") return RawChangelogViewSchema;
+
+  return RawChangelogJSONViewSchema;
+};
+
+export const SelectRawChangelogSchema = (type = "STRING") => {
+  if (type === "STRING") return RawChangelogSchema;
+
+  return RawChangelogJSONSchema;
+};
+
 /*
  * We cannot specify a schema for view creation, and all view columns default
  * to the NULLABLE mode.
  */
 
 export const RawChangelogViewSchema = {
   fields: [
+    ...defaultViewSchemaFields,
     {
-      name: "timestamp",
-      mode: "NULLABLE",
-      type: "TIMESTAMP",
-      description:
-        "The commit timestamp of this change in Cloud Firestore. If the operation is IMPORT, this timestamp is epoch to ensure that any operation on an imported document supersedes the IMPORT.",
-    },
-    {
-      name: "event_id",
-      mode: "NULLABLE",
-      type: "STRING",
-      description:
-        "The ID of the most-recent document change event that triggered the Cloud Function created by the extension. Empty for imports.",
-    },
-    {
-      name: "document_name",
+      name: "data",
       mode: "NULLABLE",
       type: "STRING",
       description:
-        "The full name of the changed document, for example, projects/collection/databases/(default)/documents/users/me).",
-    },
-    {
-      name: "operation",
-      mode: "NULLABLE",
-      type: "STRING",
-      description: "One of CREATE, UPDATE, IMPORT.",
+        "The full JSON representation of the current document state.",
     },
+    documentIdField,
+  ],
+};
+
+export const RawChangelogJSONViewSchema = {
+  fields: [
+    ...defaultViewSchemaFields,
     {
       name: "data",
```

Comment:
The field `path_params` should also be made a JSON object depending on the users selection.